### DISCUSSION TITLE
How to make use of the webpack configuration defaults in a plugin context

### DISCUSSION THREAD

**Discussion by Den-dp** [webpack/webpack#13387] [Q&A] [answered]
I like the concept of "sensible defaults" used in webpack, which helps to write less, relying on conventions.
That's why I'm trying to make use of these webpack configuration defaults in plugins as much as possible.

But right now I can't come up with an easy solution for getting those in the context of webpack plugin.

Let's say we have a plugin that uses `output.path` value.

What will be the simplest way of getting this value in the plugin, assuming that webpack@5 users can omit setting [output.path](https://webpack.js.org/migrate/5/#clean-up-configuration) value if it's equal to `path.resolve(__dirname, 'dist')`?

Should I also provide this convention in my plugin by hardcoding `path.resolve(__dirname, 'dist')` for cases when `output.path` isn't defined, or am I missing something?

**Comment by alexander-akait** [comment, marked as answer]
`compiler.outputPath` https://github.com/webpack/webpack/blob/master/types.d.ts#L1818

**Reply by Den-dp** [reply]
@alexander-akait thanks for the quick response!

So what I tried right now is the following
```
npm init -y
npm i webpack webpack-cli
```
```
module.exports = {
   plugins: [
        new class ExamplePlugin{
            apply(compiler) {
                console.log(compiler.outputPath);
            }
        }
    ]
};
```
And fount that it's an empty string

![image](https://user-images.githubusercontent.com/1770529/118362788-b66aac00-b599-11eb-8b6b-dd62959c870f.png)

So `apply` method isn't a good place to read `compiler.outputPath`?

**Reply by alexander-akait** [reply]
It will be default after this hook https://github.com/webpack/webpack/blob/master/lib/webpack.js#L79

**Reply by Den-dp** [reply]
Oh, nvm, now I get it
![image](https://user-images.githubusercontent.com/1770529/118363051-3a716380-b59b-11eb-8161-3da2b522f27d.png)
Thanks for helping me
